sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,162 @@
+            """
+
+            for regex in self.collect_regexes:
+                if re.match(regex, namespace):
+                    return True
+            return False
+
+        self.collect_regexes = self._setup_namespace_regexes()
+
+        return list(set([n for n in nsps if _match_namespace(n)]))
+
+    def _get_namespaces(self):
+        res = self.exec_cmd(
+            'microshift get namespaces'
+            ' -o custom-columns=NAME:.metadata.name'
+            ' --no-headers')
+        if res['status'] == 0:
+            return self._reduce_namespace_list(res['output'].split('\n'))
+        return []
+
+    def _get_cluster_resources(self):
+        """Get cluster-level (non-namespaced) resources to collect
+        """
+        global_resources = [
+            'apiservices',
+            'certificatesigningrequests',
+            'clusterrolebindings',
+            'clusterroles',
+            'componentstatuses',
+            'csidrivers',
+            'csinodes',
+            'customresourcedefinitions',
+            'flowschemas',
+            'ingressclasses',
+            'logicalvolumes',
+            'mutatingwebhookconfigurations',
+            'nodes',
+            'persistentvolumes',
+            'priorityclasses',
+            'prioritylevelconfigurations',
+            'rangeallocations',
+            'runtimeclasses',
+            'securitycontextconstraints',
+            'selfsubjectaccessreviews',
+            'selfsubjectrulesreviews',
+            'storageclasses',
+            'subjectaccessreviews',
+            'tokenreviews',
+            'validatingwebhookconfigurations',
+            'volumeattachments'
+        ]
+
+        _filtered_resources = []
+
+        for resource in global_resources:
+            res = self.exec_cmd("microshift get %s" % resource)
+            if res['status'] == 0:
+                _filtered_resources.append(resource)
+        return _filtered_resources
+
+    def setup(self):
+        """The setup() phase of this plugin will first gather system
+        information and then iterate through all default namespaces, and/or
+        those specified via the `add-namespaces` and `only-namespaces` plugin
+        options. Both of these options accept shell-style regexes.
+
+        Output format for this function is based on `oc adm inspect` command,
+        which is used to retrieve all API resources from the cluster.
+        """
+        self.add_forbidden_path('/var/lib/microshift')
+
+        _cluster_resources_to_collect = ",".join(
+            self._get_cluster_resources())
+        _namespaces_to_collect = " ".join(
+            ['ns/%s' % n for n in self._get_namespaces()])
+
+        if self.is_service_running(Microshift.plugin_name):
+            _subdir = self.get_cmd_output_path(make=False)
+            _kubeconfig = self.get_option('kubeconfig')
+            self.add_cmd_output(
+                'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
```

Comment:
I'm a bit worried that creating `microshift inspect` as an API forces us down a specific implementation path that will be difficult to change in the future. What's the best way to maintain some flexibility there?

Should the command be invoked with its full path, instead of relying on $PATH?